kotest-framework/kotest-framework-plugin-gradle/src/main/kotlin/io/kotest/framework/gradle/tasks/KotestJvmTask.kt

```diff
@@ -1,24 +1,21 @@
-package io.kotest.framework.gradle
+package io.kotest.framework.gradle.tasks
 
+import io.kotest.framework.gradle.TestLauncherExecBuilder
 import org.gradle.api.GradleException
-import org.gradle.api.internal.file.FileCollectionFactory
-import org.gradle.api.internal.file.FileResolver
 import org.gradle.api.plugins.JavaPluginExtension
 import org.gradle.api.tasks.CacheableTask
 import org.gradle.api.tasks.TaskAction
-import org.gradle.internal.concurrent.ExecutorFactory
+import org.gradle.process.ExecOperations
 import javax.inject.Inject
 
 // gradle requires the class be extendable
 @CacheableTask // this allows gradle to cache our inputs
-open class KotestJvmTask @Inject constructor(
-   private val fileResolver: FileResolver,
-   private val fileCollectionFactory: FileCollectionFactory,
-   private val executorFactory: ExecutorFactory,
+abstract class KotestJvmTask @Inject internal constructor(
+   private val executors: ExecOperations,
 ) : AbstractKotestTask() {
 
    @TaskAction
-   fun execute() {
+   protected fun execute() {
```

Comment:
what does this do ?

Reply:
It just keeps the API surface slightly smaller, so should someone configure the task in a build script it's not presented as an option.

Reply:
makes sense